plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -38,11 +39,19 @@
             Domain.all(domain.getType()),
             domain);
 
+    PredicatePushdownController PUSHDOWN_AND_KEEP = (session, domain) -> new DomainPushdownResult(
+            domain.simplify(getDomainCompactionThreshold(session)),
+            domain);
+
     PredicatePushdownController CASE_INSENSITIVE_CHARACTER_PUSHDOWN = (session, domain) -> {
         checkArgument(
                 domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
                 "CASE_SENSITIVE_PUSHDOWN can be used only for chars and varchars");
 
+        if (isCaseInsensitivePredicatePushdownEnabled(session)) {
+            return PUSHDOWN_AND_KEEP.apply(session, domain);
```

Comment:
or FULL_PUSHDOWN. I see one could argue both ways. 

maybe add a comment and we may need to revisit this later, to unlock e.g. agg or join pushdown on top of varchar columns